pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -198,12 +198,112 @@ func (c *Cluster) validateOSDSettings() error {
 	return nil
 }
 
+func (c *Cluster) ValidateTopologyAcrossNodes() error {
+	nodes, err := k8sutil.GetKubernetesNodesMatchingRookNodes(c.clusterInfo.Context, c.spec.Storage.Nodes, c.context.Clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to list nodes for topology validation")
+	}
+
+	return checkTopologyConflicts(nodes)
+}
+
+func checkTopologyConflicts(nodes []corev1.Node) error {
+	hierarchy := []string{
+		"kubernetes.io/hostname",
+		"topology.kubernetes.io/region",
+		"topology.kubernetes.io/zone",
+		"topology.rook.io/datacenter",
+		"topology.rook.io/room",
+		"topology.rook.io/pod",
+		"topology.rook.io/pdu",
+		"topology.rook.io/row",
+		"topology.rook.io/rack",
+		"topology.rook.io/chassis",
+	}
+
```

Comment:
How about an info log statement that the node topology is being checked?

Reply:
done

Reply:
done